### ISSUE TITLE
Does not run after reboot

### ISSUE THREAD

**Issue by HammerDownT680** [unifi-utilities/unifios-utilities#390]
Going nuts trying to get UDM-Boot to run my igmpproxy script. 

Unifi OS UDM Pro 1.12.22
Network 7.1.67

Steps: 

1) Have reinstalled udm-boot multiple times using the manual and automated commands
2) Manually enabled udm-boot using the systemctl enable commands
3) Everything enables fine when first being setup and it all works, however after a reboot it all goes horribly wrong. I have to   manually run the igmp proxy script to get my iptv running again. 
4) Each script runs just fine by themselves when manually running them
5) Tried updating to the latest Unify OS (1.12.22 as it did not work under previous versions either)

Any help would be greatly appreciated. I can't wrap my head around what is happening or how to fix it. I am not clear on what exit status 124 means, or what xarg 255 is happening. 

######### AFTER REBOOT THE FOLLOWING IS THE STATUS OF UDM-BOOT #######

# unifi-os shell
root@ubnt:/# systemctl status udm-boot
● udm-boot.service - Run On Startup UDM
   Loaded: loaded (/lib/systemd/system/udm-boot.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Mon 2022-07-11 00:16:32 PDT; 1min 50s ago
 Main PID: 77 (code=exited, status=124)
      CPU: 102ms

Jul 11 00:16:31 ubnt systemd[1]: Starting Run On Startup UDM...
Jul 11 00:16:32 ubnt systemd[1]: udm-boot.service: Current command vanished from the unit file, execution of the command list won't be resumed.
Jul 11 00:16:32 ubnt ssh-proxy[77]: udm-boot.service: running /mnt/data/on_boot.d/99-run-igmpproxy.sh
Jul 11 00:16:32 ubnt ssh-proxy[77]: xargs: sh: exited with status 255; aborting
Jul 11 00:16:32 ubnt systemd[1]: udm-boot.service: Main process exited, code=exited, status=124/n/a
Jul 11 00:16:32 ubnt systemd[1]: udm-boot.service: Failed with result 'exit-code'.
Jul 11 00:16:32 ubnt systemd[1]: Failed to start Run On Startup UDM.
Jul 11 00:16:32 ubnt systemd[1]: udm-boot.service: Consumed 102ms CPU time.

########### THE SCRIPT I AM TRYING TO RUN THAT IS PLACED IN THE NON SHELL UNDER /MNT/DATA/ON_BOOT.D/

File: 99-run-igmpproxy.sh

# cat 99-run-igmpproxy.sh 
#!/bin/sh
/mnt/data/igmpproxy/igmpproxy.sh /mnt/data/igmpproxy/igmpproxy.conf

** Have tried variations as well including removing the full path of the config

**Comment by HammerDownT680** [comment]
Immediately after installing manually it works successfully but still does not run after a reboot. I can restart the udm-boot service then it works fine. Its almost as if it needs a delay before starting due to some xarg thing that is beyond my understanding. 

root@ubnt:/# systemctl status udm-boot
● udm-boot.service - Run On Startup UDM
   Loaded: loaded (/lib/systemd/system/udm-boot.service; enabled; vendor preset: enabled)
   Active: active (exited) since Mon 2022-07-11 14:09:41 PDT; 7s ago
  Process: 5934 ExecStart=/sbin/ssh-proxy mkdir -p /mnt/data/on_boot.d && find -L /mnt/data/on_boot.d -iname "*.sh" -mindepth 1 -maxdepth 1 -type f -print0 |
 sort -z | xargs -0 -r -n 1 -- sh -c 'if test -x "$0"; then echo "udm-boot.service: running $0"; "$0"; else case "$0" in *.sh) echo "udm-boot.service: sourci
ng $0"; . "$0";; *) echo "udm-boot.service: ignoring $0";; esac; fi' (code=exited, status=0/SUCCESS)
 Main PID: 5934 (code=exited, status=0/SUCCESS)
      CPU: 35ms

Jul 11 14:09:37 ubnt systemd[1]: Starting Run On Startup UDM...
Jul 11 14:09:38 ubnt ssh-proxy[5934]: udm-boot.service: running /mnt/data/on_boot.d/05-install-cni-plugins.sh
Jul 11 14:09:39 ubnt ssh-proxy[5934]: Pouring /mnt/data/.cache/cni-plugins/cni-plugins-linux-arm64-v1.1.1.tgz
Jul 11 14:09:41 ubnt ssh-proxy[5934]: udm-boot.service: running /mnt/data/on_boot.d/06-cni-bridge.sh
Jul 11 14:09:41 ubnt ssh-proxy[5934]: udm-boot.service: running /mnt/data/on_boot.d/99-run-igmpproxy.sh
Jul 11 14:09:41 ubnt systemd[1]: Started Run On Startup UDM.

**Comment by boostchicken** [comment]
So it's not xargs, whatever script you are using is causing sh (the shell) to return 255.  What are the contents of your script?

**Comment by HammerDownT680** [comment]
Thanks for the reply! Greatly appreciate it. My script (07-igmpproxy.sh) only has 2 lines and is as follows:

#!/bin/sh
/mnt/data/igmpproxy/igmpproxy /mnt/data/igmpproxy/igmpproxy.conf

**Comment by boostchicken** [comment]
What other scripts are you running. Also where did you get the binary you installed from.  Please make sure it is the latest.

**Comment by HammerDownT680** [comment]
This is the only script I am running except for the 2 CNI scripts the come with this installation. I have removed everything and reinstalled the latest binary. Again it works immediately however upon restart I receive the same error.

**Comment by boostchicken** [comment]
does the CNI stuff run on reboot?

**Comment by HammerDownT680** [comment]
It appears to. Seems that the error occurs after my one line script. I don't receive errors if I remove my script

**Comment by boostchicken** [comment]
delete your script, start fresh, might have some crazy white space or something in there?

**Comment by royord** [comment]
I too am having issues the boot script, I have deleted my script and brought it back without a change in functionality.

**Comment by HammerDownT680** [comment]
Yeah I could not get it to work no matter how I changed my script. I ended up using fapienishere's UDM-iptv script instead of my own script. It works on boot just fine alongside this one.

**Comment by royord** [comment]
I'm trying to keep a VPN running on reboots using method for openvpn, it's just the fact that I need a script to run afterwards to re-symlink a directory to /etc/ to make everything work properly...

https://github.com/peacey/split-vpn#how-do-i-run-this-at-boot

**Comment by HammerDownT680** [comment]
Ah ok. Can you go into UniFi-os shell and run systemctl status udm-boot.service and provide the output?

**Comment by royord** [comment]
```
● udm-boot.service - Run On Startup UDM
   Loaded: loaded (/lib/systemd/system/udm-boot.service; disabled; vendor preset: enabled)
   Active: inactive (dead)
```

**Comment by HammerDownT680** [comment]
Make sure you enable the service "systemctl enable udm-boot" and ensure that your script has been chmod +x to enable execution. Any logs indicating activity?

**Comment by Tntdruid** [comment]
`unifi-os shell 
sudo systemctl enable udm-boot.service 
sudo systemctl start udm-boot.service 
exit 
reboot`

**Comment by royord** [comment]
@boostchicken I'm not sure that the original question above for @HammerDownT680 was answered...
